RationaleMCP/0031/source-locations.md

```diff
@@ -0,0 +1,77 @@
+# Source locations
+
+To fully serve as a format for intermediate representation of translation of full Modelica, the Flat Modelica code can be decorated with source location information pointing back to the original full Modelica model (or other source of the Flat Modelica code).
+
+
+## Design goals
+
+Some design goals to keep in mind when comparing the design alternatives:
+- The detailed format of a source location needs to be flexible to serve the needs of different tools.
+- A tool should be able to ignore source locations originating from other tools.
+- Source locations may be needed on many levels of the Flat Modelica code, including component declarations, equations, modifications, and expressions.
+- Reduced Flat Modelica soruce code human readability is a concern when soruce locations are added.
+- Simple compression of the source locations could be handy.
+
```

Comment:
Web Meeting (Hans, Henrik, Gerd, Martin, Oliver)

1. What kind of information has to be captured?

- Reference to the source file (filename, line, column) or instance tree (fully qualified name)
- Could be anything, not to be decided now for prototyping.
- There must be the option to express vendor specific stuff, e.g. transformations.

2. Where should the information be stored?

- External file, referenced by an annotation on model level.
```
package 'mymodel'
model 'mymodel'
...
end 'mymodel';
annotation(SourceLocations(...));
end 'mymodel';
```
- inline, special kind of annotation 

4. How to attach the information in the code?
5. How to express the associated information?